src/test/java/com/connexta/ingest/service/impl/IngestServiceImplUnitTests.java

```diff
@@ -0,0 +1,322 @@
+
+    // then
+    final InOrder inOrder =
+        Mockito.inOrder(mockStoreClient, mockMetacardStorageAdaptor, mockTransformClient);
+    // TODO Verify the key was generated as expected instead of anyString() and any(URI.class) here.
+    // Maybe extract a key generator.
+    inOrder
+        .verify(mockMetacardStorageAdaptor)
+        .store(eq(metacardFileSize), eq(metacardInputStream), anyString());
+    inOrder
+        .verify(mockTransformClient)
+        .requestTransform(eq(location), eq(mimeType), any(URI.class));
+  }
+
+  /* START ingest StoreClient tests */
+
+  @Test
+  public void testIngestWhenStoreClientThrowsStoreException() {
+    // given
+    final IngestService ingestService =
+        new IngestServiceImpl(
+            mockStoreClient,
+            mockMetacardStorageAdaptor,
+            "http://localhost:9040/ingest/",
+            mockTransformClient);
+
+    final Long fileSize = 1L;
+    final String mimeType = "mimeType";
+    final InputStream inputStream = mock(InputStream.class);
+    final String fileName = "fileName";
+    final StoreException storeException = mock(StoreException.class);
+    when(mockStoreClient.store(fileSize, mimeType, inputStream, fileName))
+        .thenThrow(storeException);
+
+    // when
+    final StoreException thrown =
+        assertThrows(
+            StoreException.class,
+            () ->
+                ingestService.ingest(
+                    fileSize, mimeType, inputStream, fileName, 1L, mock(InputStream.class)));
+
+    // then
+    assertThat(thrown, is(storeException));
+  }
+
+  @Test
+  public void testIngestWhenStoreClientThrowsThrowable() {
+    // given
+    final IngestService ingestService =
+        new IngestServiceImpl(
+            mockStoreClient,
+            mockMetacardStorageAdaptor,
+            "http://localhost:9040/ingest/",
+            mockTransformClient);
+
+    final Long fileSize = 1L;
+    final String mimeType = "mimeType";
+    final InputStream inputStream = mock(InputStream.class);
+    final String fileName = "fileName";
+    final Throwable throwable = mock(StoreException.class);
+    when(mockStoreClient.store(fileSize, mimeType, inputStream, fileName)).thenThrow(throwable);
+
+    // when
+    final Throwable thrown =
+        assertThrows(
+            Throwable.class,
+            () ->
+                ingestService.ingest(
+                    fileSize, mimeType, inputStream, fileName, 1L, mock(InputStream.class)));
+
+    // then
+    assertThat(thrown, is(throwable));
+  }
+
+  /* END ingest StoreClient tests */
+
+  /* START ingest MetacardStorageAdaptor tests */
+
+  @Test
+  public void testIngestWhenMetacardStorageAdaptorThrowsStoreMetacardException() throws Exception {
+    // given
+    final IngestService ingestService =
+        new IngestServiceImpl(
+            mockStoreClient,
+            mockMetacardStorageAdaptor,
+            "http://localhost:9040/ingest/",
+            mockTransformClient);
```

Comment:
:warning: These tests have a lot of duplicated code

Reply:
I prefer strict mocking. If the ingest request code was extracted out, we would be doing some unnecessary mocking for some of the tests. I believe we need to develop our own tool for sending ingest requests that can be released with the ingest API which we can use in these tests, so this issue will be OBE.